src/index.js

```diff
@@ -0,0 +1,114 @@
+    this.racingCountSubmitButton.classList.add(HIDDEN_CLASSNAME);
+    this.racingCountTitle.classList.add(HIDDEN_CLASSNAME);
+    this.winnerOuputElement.classList.add(HIDDEN_CLASSNAME);
+  }
+
+  createResultElement() {
+    this.resultTitle = document.createElement("span");
+    this.resultTitle.innerHTML = "최종 우승자: ";
+    this.resultTitle.classList.add(HIDDEN_CLASSNAME);
+
+    this.resultText = document.createElement("span");
+    this.resultText.id = "racing-winners";
+    this.resultText.innerHTML = "";
+    this.resultText.classList.add(HIDDEN_CLASSNAME);
+  }
+
+  addEvent = function () {
+    this.carNameSubmitButton.addEventListener("click", (event) =>
+      this.onCarNameSubmitted(event)
+    );
+    this.racingCountSubmitButton.addEventListener("click", (event) =>
+      this.onRaceCountSubmitted(event)
+    );
+  };
+
+  onCarNameSubmitted = function (event) {
+    event.preventDefault();
+    const carNames = this.carNameInput.value.split(",");
+    const { valid, message } = checkNameValidation(carNames);
+    valid ? this.makeCarWithName(carNames) : alert(message);
+  };
+  makeCarWithName = function (carNames) {
+    carNames.forEach((name) => this.cars.push(new Car(name)));
+    this.racingCountTitle.classList.remove(HIDDEN_CLASSNAME);
+    this.racingCountInput.classList.remove(HIDDEN_CLASSNAME);
+    this.racingCountSubmitButton.classList.remove(HIDDEN_CLASSNAME);
+  };
+
+  onRaceCountSubmitted = function (event) {
+    event.preventDefault();
+    const raceCount = this.racingCountInput.value;
+    const { valid, message } = checkRaceCountValidation(raceCount);
+    valid ? this.play(raceCount) : alert(message);
+  };
+  play = function (raceCount) {
+    this.raceCount = Number(raceCount);
+    this.cars.forEach((car) =>
+      car.move(
+        RACE_GAME_INPUT.MIN,
+        RACE_GAME_INPUT.MAX,
+        RACE_GAME_INPUT.BASE,
+        this.raceCount
+      )
+    );
+    this.showTotalRecords();
+    this.showWinner();
+  };
+
+  showTotalRecords = function () {
+    let resultText = `📄 실행 결과<br/><br/>`;
+
+    const range = [...Array(Number(this.raceCount))].map((v, i) => i);
+    range.forEach((item, index) => (resultText += this.getRaceRecord(index)));
+    this.winnerOuputElement.innerHTML = `${resultText}`;
+    this.winnerOuputElement.classList.remove(HIDDEN_CLASSNAME);
+  };
+
+  getRaceRecord = function (lap) {
+    const record = [];
+    this.cars.forEach((car) =>
+      record.push(`${car.name}:${" -".repeat(car.result[lap])}`)
+    );
+    record.push("<br/>");
+    return record.join("<br/>");
+  };
+
+  showWinner = function () {
+    const winner = [];
+    this.cars.forEach((car) =>
+      car.result[this.raceCount - 1] === this.raceCount
+        ? winner.push(car.name)
+        : ""
+    );
+    this.resultText.innerHTML = `${winner.join(", ")}`;
+    this.winnerOuputElement.appendChild(this.resultTitle);
+    this.winnerOuputElement.appendChild(this.resultText);
```

Comment:
```suggestion
    this.winnerOuputElement.append(this.resultTitle, this.resultText);
```

`appendChild` 와 `append` 의 차이는 무엇일까요?

Reply:
append : 노드 객체, 문자열 모두 추가할 수 있고 여러 개 한번에 추가 가능
appendChild : 노드 객체만 추가할 수 있고 하나만 추가 가능
---
같은 부모에 추가하는 것이니 append로 한꺼번에 추가하는 것이 더 좋았겠군요! ㅎㅎ

Reply:
>  문자열 모두 추가 
:+1: :+1: :+1: :+1: 

추가로 `appendChild` 에서는 문자열을 어떻게 추가하지? 라는 생각도 한번 해보셨으면 좋겠네요 :+1: